src/Modules/Workspaces/Commands.Workspaces/GetPowerBIWorkspace.cs

```diff
@@ -76,10 +92,20 @@ protected override void ExecuteCmdlet()
                 this.Filter = string.IsNullOrEmpty(this.Filter) ? OrphanedFilterString : $"({this.Filter}) and ({OrphanedFilterString})";
             }
 
-            if(this.Id != default && this.Scope == PowerBIUserScope.Organization)
+            if(this.ParameterSetName == IdParameterSetName && this.Scope == PowerBIUserScope.Organization)
+            {
+                this.Filter = $"id eq '{this.Id}'";
+            }
+
+            if (this.ParameterSetName == NameParameterSetName && this.Scope == PowerBIUserScope.Organization)
+            {
+                this.Filter = $"tolower(name) eq '{this.Name.ToLower()}'";
+            }
+
+            if(!string.IsNullOrEmpty(this.User) && this.Scope == PowerBIUserScope.Organization)
             {
-                var idFilter = $"id eq '{this.Id}'";
-                this.Filter = string.IsNullOrEmpty(this.Filter) ? idFilter : $"({idFilter}) and ({this.Filter})";
+                var userFilter = $"users/any(u: tolower(u/emailAddress) eq '{this.User.ToLower()}')";
```

Comment:
Consider adding unit tests for the new functionality of this cmdlet as you have time.

Reply:
I added one :)